spec/support/authentication.rb

```diff
@@ -50,3 +56,39 @@ def create_user(login)
   }
   let(:api_key) { the_user.credentials.api_key }
 end
+
+shared_context "create host" do
+  def create_host(host_login)
+    id = "rspec:host:#{host_login}"
+    host_role = Role.create(role_id: id).tap do |role|
+      options = { role: role }
+      Credentials.create(options)
+      role.reload
+    end
+    Resource.create(resource_id: id, owner_id: id).tap do |resource|
+      resource.reload
+      host_role.reload
+    end
+
+    return host_role
+  end
+
+  let(:host_login) { "default-host-login" }
+  # 'let!' always runs before the example; 'let' is lazily evaluated.
+  let!(:the_host) {
+    create_host(host_login)
+  }
```

Comment:
2 small things here:
- Formatting nit: If it can fit on 1 line within 80 chars, use 1 line.
- I would in general avoid all `!` methods in ruby, including this rspec DSL one.  If you need something done sequentially, move the statement into the test case itself.  The will make the logic more explicit.

Reply:
sorry - was following the existing code in the test.  moved into the actual spec file